compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaTypes.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Optional<Symbol> getTypeByName(String org, String moduleName, String version, String typeDefName) {
+        if (org == null || moduleName == null || version == null || typeDefName == null) {
+            throw new IllegalArgumentException("Null parameters are not allowed. Found parameter values are org: " +
+                    org + " moduleName: " + moduleName + ", version: " + version + ", and typeDefName: " + typeDefName);
+        }
+
+        PackageID packageID = new PackageID(Names.fromString(org), Names.fromString(moduleName),
+                Names.fromString(version));
+
+        BPackageSymbol packageSymbol = packageCache.getSymbol(packageID);
+        if (packageSymbol == null) {
+            return Optional.empty();
+        }
+
+        SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+        Scope.ScopeEntry entry = pkgEnv.scope.lookup(Names.fromString(typeDefName));
+        if (isValidTypeDef(entry.symbol)) {
+            return Optional.of(symbolFactory.getBCompiledSymbol(entry.symbol, typeDefName));
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Optional<Map<String, Symbol>> typesInModule(String org, String moduleName, String version) {
+        if (org == null || moduleName == null || version == null) {
+            throw new IllegalArgumentException("Null parameters are not allowed. Found parameter values are org: " +
+                    org + " moduleName: " + moduleName + ", and version: " + version);
+        }
+
+        PackageID packageID = new PackageID(Names.fromString(org), Names.fromString(moduleName),
+                Names.fromString(version));
+
+        BPackageSymbol packageSymbol = packageCache.getSymbol(packageID);
+        if (packageSymbol == null) {
+            return Optional.empty();
+        }
+
+        SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+        Scope pkgEnvScope = pkgEnv.scope;
+
+        if (pkgEnvScope == null || pkgEnvScope.entries == null) {
+            return Optional.empty();
+        }
+
+        Map<String, Symbol> typeDefSymbols = new FieldMap<>();
+        for (Scope.ScopeEntry scopeEntry : pkgEnvScope.entries.values()) {
+            BSymbol bSymbol = scopeEntry.symbol;
+            if (isValidTypeDef(bSymbol)) {
+                String typeDefName = bSymbol.getOriginalName().getValue();
+                typeDefSymbols.put(typeDefName, symbolFactory.getBCompiledSymbol(bSymbol, typeDefName));
+            }
+        }
+
+        return Optional.of(Collections.unmodifiableMap(typeDefSymbols));
+    }
+
+    private boolean isValidTypeDef(BSymbol bSymbol) {
+        return bSymbol != null && bSymbol.getOrigin() != SymbolOrigin.VIRTUAL
+                && (bSymbol.getKind() == SymbolKind.TYPE_DEF
+                || bSymbol.getKind() == SymbolKind.CONSTANT
+                || bSymbol.getKind() == SymbolKind.ENUM
```

Comment:
We are considering enum members as a type too right? If so, need to consider that here.

Reply:
Ah no.. in `BSymbol` enum members are just constant symbols I guess 🤔

Reply:
@pubudu91 
Yes, enum members are just constants and covered by test cases as well.